tests/conftest.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+import sys
+import subprocess
+
+import pytest
+from six import StringIO
+
+from util import SphinxTestApp, path
+
+
+@pytest.fixture
+def app_params(request):
+    """
+    parameters that is specified by 'pytest.mark.sphinx' for
+    sphinx.application.Sphinx initialization
+    """
+    markers = request.node.get_marker("sphinx")
+    pargs = {}
+    kwargs = {}
+
+    if markers is not None:
+        # to avoid stacking positional args
+        for info in reversed(list(markers)):
+            for i, a in enumerate(info.args):
+                pargs[i] = a
+            kwargs.update(info.kwargs)
+
+    args = [pargs[i] for i in sorted(pargs.keys())]
+    return args, kwargs
+
+
+@pytest.fixture(scope='function')
+def app(app_params, make_app):
+    """
+    provides sphinx.application.Sphinx object
+    """
+    args, kwargs = app_params
+    app_ = make_app(*args, **kwargs)
+    yield app_
+
+
+@pytest.fixture(scope='function')
+def status(app):
+    """
+    compat for testing with previous @with_app decorator
+    """
+    return app._status
+
+
+@pytest.fixture(scope='function')
+def warning(app):
+    """
+    compat for testing with previous @with_app decorator
+    """
+    return app._warning
+
+
+@pytest.fixture()
+def make_app():
+    """
+    provides make_app function to initialize SphinxTestApp instance.
+    if you want to initialize 'app' in your test function. please use this
+    instead of using SphinxTestApp class directory.
+    """
+    apps = []
+    syspath = sys.path[:]
+
+    def make(*args, **kwargs):
+        status, warning = StringIO(), StringIO()
+        kwargs.setdefault('status', status)
+        kwargs.setdefault('warning', warning)
+        app_ = SphinxTestApp(*args, **kwargs)
+        apps.append(app_)
+        return app_
+    yield make
+
+    sys.path[:] = syspath
```

Comment:
In current (nose) testing, some sphinx project for testing insert path on conf.py. I think developers can't do cleanup correctly for such case because it is difficult to notice.